### ISSUE TITLE
Rule Change: (require-await at inherited class methods)

### ISSUE THREAD

**Issue by zuhairtaha** [eslint/eslint#16141]
### What rule do you want to change?

require-await

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
class A {
  /**
   * @return {!Promise<!string[]>}
   */
  async getNames() {
    const res = await fetch("https://jsonplaceholder.typicode.com/users");
    const users = await res.json();
    return users.map(user => user.name);
  }
}

class B extends A {
  /**
   * @return {!Promise<!string[]>}
   */
  async getNames() { // Async method 'getNames' has no 'await' expression.eslint require-await
    return ["John", "Jane"];
  }
}

const a = new A();
const b = new B();
```

### What does the rule currently do for this code?

At class `B` it shows an error on the method `getNames()`.

### What will the rule do after it's changed?

It should not show an error on the extended class methods of async methods, or the extended class of abstract

If I create a new instance of A, or B and want to call `getNames()` it should be in both the same, I mean both async
even if there is no await. Also for abstract classes

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
Unfortunately, this is beyond ESLint’s capabilities. Tracking classes and subclasses requires some type checking data that ESLint does not have access to. It treats each method as if it were standalone. In this case, your best option is to use a disable comment for the method in question.
